### ISSUE TITLE
add vector methods for whole and decimal numbers in `standalone-types-check.R`

### ISSUE THREAD

**Issue by EmilHvitfeldt** [r-lib/rlang#1714]
In much the same way `check_character()` is the vector version of `check_string()`, and `check_logical()` is the vector version of `check_bool()`. It would be nice to have the same for `check_number_whole()` and `check_number_decimal()`

I would be happy to take stab at this

**Comment by lionel-** [comment]
With plural names like `check_numbers_whole()`? This seems reasonable, a PR is welcome! cc @hadley

**Comment by ijlyttle** [comment]
Would you consider `check_numeric_whole()`? It could fit a pattern of adjectives used for the vector version, and nouns for the  scalar version (I know "character" is a stretch)

**Comment by hadley** [comment]
I think I'd prefer `check_double()` and `check_integer()` although this might be overly abusing the difference between what people mostly mean when they say integer vector and the precise technical definition.

**Comment by lionel-** [comment]
@hadley We already have the singular `check_number_whole()` and `check_number_decimal()`

**Comment by hadley** [comment]
@lionel- right, but I think vector and scalar checks are different enough that they can use different names/conventions (e.g. string vs character, bool vs logical).

**Comment by lionel-** [comment]
oh that makes sense. But I would be very surprised if `check_double()` accepted integer-storage vectors. So I'd prefer`check_numeric_whole()` here which is more technically correct and also clearer in intent.

**Comment by lionel-** [comment]
Looking back on the existing predicates, I find it surprising that `check_number_decimal(1L)` doesn't fail.

I wonder if we should rename it to `check_number()`. "number" would be the scalar version of "numeric". And we'd have `check_number_whole()` and `check_numeric_whole()` as whole-number variants.

**Comment by khusmann** [comment]
Just submitted a PR for this, but it might be worth discussing the naming convention here a bit more:

Here's a matrix of our current `check_*()` vs `is_*()` functions:

| Type                  | Single Check        | Vector Check         | Single is   | Vector is                          |
|-----------------------|---------------------|----------------------|-------------|------------------------------------|
| numeric (double or integer) | `check_number`      | `check_numeric`       |          | `is.numeric` *(no rlang is_numeric!)* |
| whole numeric         | `check_number_whole` | `check_numeric_whole` |   `is_scalar_integerish`       | `is_integerish`                    |
| double                |                |                |       `is_scalar_double` |          `is_double`                     |
| integer               |                  |                   |   `is_scalar_integer`       | `is_integer`                       |
| string / character    | `check_string`       | `check_character`     | `is_string`, `is_scalar_character` | `is_character`                     |
| bool / logical        | `check_bool`         | `check_logical`       | `is_bool`, `is_scalar_logical`   | `is_logical`                       |

Personally, I like "integerish" instead of "numeric_whole" (Maybe it's because I come from a math background, but I expect whole number to be a non-negative integer).

What if we did something like the following? (Changes / additions in bold):

| Type                  | Single Check        | Vector Check         | Single is   | Vector is                          |
|-----------------------|---------------------|----------------------|-------------|------------------------------------|
| numeric (double or integer) | `check_number`, **`check_scalar_numeric`**      | `check_numeric`       |    **`is_number`**, **`is_scalar_numeric`**     | **`is_numeric`**  |
| whole numeric         | **`check_scalar_integerish`** | **`check_integerish`** |     `is_scalar_integerish`     | `is_integerish`                    |
| double                |      **`check_scalar_double`**          |  **`check_double`**              | `is_scalar_double` |               `is_double`                     |
| integer               |        **`check_scalar_integer`**          |  **`check_integer`**                 |    `is_scalar_integer`      | `is_integer`                       |
| string / character    | `check_string`, **`check_scalar_character`**       | `check_character`     | `is_string`, `is_scalar_character` | `is_character`                     |
| bool / logical        | `check_bool`, **`check_scalar_logical`**         | `check_logical`       | `is_bool`, is_scalar_logical`   | `is_logical`                       |

**Comment by khusmann** [comment]
> I don't think we need a check function for every is_ function since our understanding of useful types has evolved since we created the is_ functions.

Ah ok, that's key context for me; I've been treating the is_ fns as the sort of ground truth primitive definitions.

> And I don't think a check_type() function is feasible without a lot of work because it's hard to get the error messages precisely correct unless you have all of the information in one place

Excellent point. I think I'm on the same page now, thanks for taking the time to get me up to speed.
